tests/Doctrine/Tests/DBAL/Query/QueryBuilderTest.php

```diff
@@ -476,6 +476,38 @@ public function testInsertValues()
         self::assertEquals('INSERT INTO users (foo, bar) VALUES(?, ?)', (string) $qb);
     }
 
+    public function testInsertSelect()
+    {
+        $qb = new QueryBuilder($this->conn);
+        $qb->insert('users')
+            ->select('u2.name', 'u2.password')
+            ->from('users', 'u2')
+            ->where('u2.id = ?');
+
+        self::assertEquals(QueryBuilder::INSERT, $qb->getType());
+        self::assertEquals('INSERT INTO users (SELECT u2.name, u2.password FROM users u2 WHERE u2.id = ?)', (string) $qb);
+    }
+
+    public function testInsertValuesSelect()
+    {
+        $qb = new QueryBuilder($this->conn);
+        $qb->insert('users')
+            ->values(
```

Comment:
This syntax is a bit weird. As far as I understand the resulting SQL, this method defines the columns which the selected columns will be mapped to. Not values.

Reply:
that's right